packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/CustomVisConfigTabs.tsx

```diff
@@ -65,6 +157,257 @@ const loadMonaco = (monaco: Monaco, schemas: Schema[]) => {
+                        if (isDimension(item)) return 2;
+                        if (isCustomDimension(item)) return 3;
+                        if (isMetric(item)) return 4;
+                        return 5; // everything else
+                    };
+                    return getPriority(a) - getPriority(b);
+                },
+            );
+
+            /**
+             * When selecting a field for the y axis (and color/size values),
+             * we want to prioritize numeric metrics and table calculations
+             * over dimensions
+             */
+            const sortedItemsForY = Object.values(itemsMap || {}).sort(
+                (a, b) => {
+                    const getPriorityForY = (item: ItemsMap[string]) => {
+                        if (isMetric(item) && isNumericType(item.type))
+                            return 1;
+                        if (isMetric(item)) return 2;
+                        if (isTableCalculation(item)) return 3;
+                        return 4; // everything else
+                    };
+
+                    return getPriorityForY(a) - getPriorityForY(b);
+                },
+            );
+
+            const xField = sortedItemsForX[0];
+            const [yField, extraField] = sortedItemsForY;
+
+            const templateString = generateVegaTemplate(
+                template,
+                xField,
+                yField,
+                extraField,
+            );
+            setEditorConfig(templateString);
+        },
+        [isCustomConfig, itemsMap, setEditorConfig],
+    );
+    return (
+        <>
+            {/*
+            // confirmation modal ? yes or no
+            // you can always do cTRL+z to go back 
+            <Modal
+                title="Load template"
+                opened={!!selectedTemplate}
+                onClose={() => {
+                    setSelectedTemplate(undefined);
+                }}
+            >
+                <Text>
+                    Loading a new template will overwrite your current chart
+                    configuration. Are you sure you want to continue?
+                </Text>
+                <Group position="right" mt="sm">
+                    <Button
+                        color="dark"
+                        variant="outline"
+                        onClick={() => {
+                            setSelectedTemplate(undefined);
+                        }}
+                    >
+                        Keep config
+                    </Button>
+                    <Button
+                        onClick={() => {
+                            if (selectedTemplate)
+                                loadTemplate(selectedTemplate);
+                            setSelectedTemplate(undefined);
+                        }}
+                    >
+                        Load template
+                    </Button>
+                </Group>
+            </Modal>*/}
+
+            <Select
```

Comment:
I don't think `select` is the best component here, I think a `actionButton` + `popover` would be better.